rakelib/license.rake

```diff
@@ -0,0 +1,45 @@
+# we need to call exit explicity  in order to set the proper exit code, otherwise
+# most common CI systems can not know whats up with this tests.
+require_relative "default_plugins"
+
+namespace "license" do
+
+  desc "run core specs"
+  #task "core" =>  ["test:setup"] do
+  task "core" do
+    require 'yaml'
+    generate_license_information
+  end
+
+end
+
+def all_installed_gems
+  Gem::Specification.all = nil
```

Comment:
Hmm https://github.com/elastic/logstash/blob/master/spec/license_spec.rb#L24 needs names.  For this analysis, that is essentially whatever will be bundled in the distributions.  Where can one grab that information?